docs/README.md

```diff
@@ -0,0 +1,423 @@
+   // delivers the current value, automatically updated as new versions become
+   // available from the server.
+   apiKey := st.Secret("prod/myprogram/secret-1")
+
+   cli := someservice.NewClient("username", apiKey.Get())
+   // ...
+}
+```
+
+Alternatively, you can obtain a [`setec.Watcher`][setecwatcher], which combines
+a secret with a channel that notifies when a new secret value is available:
+
+```go
+// Get the handle, as before.
+apiKey := st.Watcher("prod/my-program/secret-1")
+
+// Create a client with the current value.
+cli := someservice.NewClient("username", apiKey.Get())
+
+// Make a helper that will refresh the client when the secret updates.
+// This example assumes no concurrency; you may need a lock if multiple
+// goroutines will request a client at the same time.
+getClient := func() *someservice.Client {
+   select {
+   case <-apiKey.Ready():
+     cli = someservice.NewClient("username", apiKey.Get())
+   default:
+   }
+   return cli
+}
+
+// Make API calls using the helper.
+rsp, err := getClient().Method(ctx, args)
+// ...
+```
+
+### Bootstrapping and Availability
+
+A reasonable concern when fetching secrets from a network service is what
+happens if the secrets service is not reachable when a program needs to fetch a
+secret (e.g., when it is starting up). A good answer depends on the nature of
+the program: Batch processing tools can usually afford to wait and retry until
+the service becomes available. Interactive services, by contrast, may not be
+able to tolerate waiting.
+
+A program that needs be able to start even when the secrets server is
+unavailable can trade a bit of security for availability by caching the active
+versions of the secrets it needs in a local file. When the program start or
+restarts, it can fall back to the cached values if the secrets service is not
+immediately available. The Go client library's [`setec.Store`][setecstore] type
+supports this kind of caching as an optional feature, and the same logic can be
+implemented in any language.
+
+In Go, you can enable a file cache using `setec.NewFileCache`:
+
+```go
+// Create or open a cache associated with the specified file path.
+fc, err := setec.NewFileCache("/data/secrets.cache")
+if err != nil {
+   return fmt.Errorf("creating cache: %w", err)
+}
+st, err := setec.NewStore(ctx, setec.StoreConfig{
+   Client:       setec.Client{Server: "https://secrets.example.ts.net"},
+   Secrets:      []string{"secret1", "secret2", "secret3"},
+   PollInterval: 12*time.Hour,
+   Cache:        fc,
+})
+// ...
+```
+
+With the cache enabled, the store will automatically persist new secrets
+fetched from the server into the cache as they become available. Morever, when
+the store is created, the store will not block waiting for the server if all
+the requested secrets already have a version stored in the cache. This permits
+the program to start up even if the setec server is not immediately available.
```

Comment:
Perhaps worth in this section explicitly mentioning the tradeoff that the cache places secrets in local storage, so you still get the benefits of secret updating and a consistent way to reason about secrets distribution, but robustness here trades off some security.

Reply:
I have made an attempt at this, PTAL?